src/server.rs

```diff
@@ -84,7 +84,9 @@ serializable_enum!(ResponseData,
     TextEdit([TextEdit; 1]),
     Locations(Vec<Location>),
     Highlights(Vec<DocumentHighlight>),
-    HoverSuccess(Hover)
+    HoverSuccess(Hover),
+    // Empty tuple is represented as null in JSON
+    Null(())
```

Comment:
I would prefer to send an empty struct, rather than null. I would also name it `Ack` rather than `Null`.